pkg/kubelet/kubelet.go

```diff
@@ -3076,3 +3061,79 @@ func (kl *Kubelet) warnCgroupV1Usage() {
 	}
 	metrics.CgroupVersion.Set(float64(cgroupVersion))
 }
+
+// staticPodRegistration ensures that all static pods are registered to the apiserver
+// before node became ready.
+func (kl *Kubelet) initialStaticPodsRegistration() error {
+	kl.initialStaticPodsRegisteredLock.Lock()
+	defer kl.initialStaticPodsRegisteredLock.Unlock()
+	// kubelet running in standalone mode does not register static pods to the apiserver.
+	if kl.kubeClient == nil {
+		return nil
+	}
+	// Check if initial static pod registration has already been completed.
+	if kl.initialStaticPodsRegistered {
+		return nil
+	}
+	staticPodToMirrorPodMap := kl.podManager.GetStaticPodToMirrorPodMap()
+	for _, mirrorPod := range staticPodToMirrorPodMap {
+		if mirrorPod == nil {
+			return fmt.Errorf("not all static pods are registered")
+		}
+	}
+
+	kl.initialStaticPodsRegistered = true
+	return nil
+}
+
+// Ensure Mirror Pod for Static Pod exists.
+func (kl *Kubelet) ensureMirrorPodExists(staticPod, mirrorPod *v1.Pod) {
+	if kubetypes.IsStaticPod(staticPod) {
```

Comment:
alternative to https://github.com/kubernetes/kubernetes/pull/126870/files#r1785780245 you just remove an indentantion by returning early
```suggestion
	if !kubetypes.IsStaticPod(staticPod) {
	     return
	 }
```